crates/ide/src/diagnostics/fixes.rs

```diff
@@ -144,6 +145,34 @@ impl DiagnosticWithFix for IncorrectCase {
     }
 }
 
+impl DiagnosticWithFix for ReplaceFilterMapNextWithFindMap {
+    fn fix(&self, sema: &Semantics<RootDatabase>) -> Option<Fix> {
+        let root = sema.db.parse_or_expand(self.file)?;
+        let next_expr = self.next_expr.to_node(&root);
+        let next_call = MethodCallExpr::cast(next_expr.syntax().clone())?;
+
+        let filter_map_call = MethodCallExpr::cast(next_call.receiver()?.syntax().clone())?;
+        let filter_map_name_range = filter_map_call.name_ref()?.ident_token()?.text_range();
+        let filter_map_args = filter_map_call.syntax().children().find_map(ArgList::cast)?;
+
+        let range_to_replace =
+            TextRange::new(filter_map_name_range.start(), next_expr.syntax().text_range().end());
+        let replacement = format!("find_map{}", filter_map_args.syntax().text());
+        let trigger_range = next_expr.syntax().text_range();
```

Comment:
I wonder if it would be nice to be able to trigger this assist on the `next` call as well as the `filter_map` call 🤔

Reply:
Unless I've misread your comment, I think you've got this the wrong way round - the assist trigger range is currently the entire expression up to and including the `next` call. See https://github.com/rust-analyzer/rust-analyzer/pull/7062/files#diff-56bc9a72638280b6b625e137fef3487ba1723684b4cbf6b43ab19230dcda788bR675 for an example.

Reply:
Oh right, I forgot that this contains the receiver 👍